### PR TITLE
Migrate ruleset_types to editor types in _ui

### PR DESCRIPTION
**Pull request by rowanseymour**
Adds #328

### CODE REVIEW

File: legacy/definition.go [hunk 1]
```diff
@@ -863,12 +864,13 @@ func migrateRuleSet(lang utils.Language, r RuleSet, localization flows.Localizat
 
 		// subflow rulesets operate on the child flow status
 		router = routers.NewSwitchRouter(defaultExit, "@child.status", cases, resultName)
+		uiType = "subflow"
```
Comment (rowanseymour) [inline legacy/definition.go]: @ericnewcomer not seeing a complete list of these so thus far just have:

* "subflow"
* "split_by_groups"
* "wait_for_response"
* "split_by_expression"

Comment (ericnewcomer) [reply legacy/definition.go]: I think that's the only ones we have implemented so far.

What else is there, I think this is close if not complete:
split_by_sms_form, split_by_contact_field, split_by_flow_result

Comment (rowanseymour) [reply legacy/definition.go]: after discussion, we're using these names:

* wait_for_response
* subflow
* split_by_groups
* split_by_expression
* split_by_contact_field
* split_by_run_result
* split_by_run_result_delimited

File: legacy/ui.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	UINodeTypeSubflow                   UINodeType = "subflow"
+	UINodeTypeWebhook                   UINodeType = "webhook"
+	UINodeTypeResthook                  UINodeType = "resthook"
+	UINodeTypeSplitByGroups             UINodeType = "split_by_groups"
+	UINodeTypeSplitByExpression         UINodeType = "split_by_expression"
+	UINodeTypeSplitByContactField       UINodeType = "split_by_contact_field"
+	UINodeTypeSplitByRunResult          UINodeType = "split_by_run_result"
+	UINodeTypeSplitByRunResultDelimited UINodeType = "split_by_run_result_delimited"
+	UINodeTypeSplitByRandom             UINodeType = "split_by_random"
+)
```
Comment (rowanseymour) [inline legacy/ui.go]: @ericnewcomer as discussed but also added resthook and random

**Pull request merged**